surfactant/cmd/tui.py

```diff
@@ -437,6 +454,91 @@ def load(self):
             self.mount(entry, before="#add_context_entry")
 
 
+class PluginSetting(textual.widgets.Static):
+    __config_manager = surfactant.configmanager.ConfigManager()
+
+    def __init__(self, plugin_name: str, info: surfactant.utils.get_plugin_settings.PluginSetting):
+        super().__init__()
+        self.plugin_name = plugin_name
+        self.info = info
+        if self.info.type_ == "str":
+            self.input_field = textual.widgets.Input()
+            default_value = self.info.default
+            if default_value is None:
+                default_value = ""
+            self.value = self.__config_manager.get(self.plugin_name, self.info.name, default_value)
+        elif self.info.type_ == "bool":
+            self.input_field = textual.widgets.Checkbox()
+            default_value = self.info.default
+            if default_value is None:
+                default_value = True
+            else:
+                # Have to convert from string to Boolean
+                default_value = default_value.lower() == "true"
+            self.value = self.__config_manager.get(self.plugin_name, self.info.name, default_value)
+        else:
+            raise TypeError(f'Invalid plugin setting of type "{self.info.type_}"')
+
+    def compose(self) -> textual.app.ComposeResult:
+        # Set the value now - setting the Input value during __init__ was causing errors...
+        self.input_field.value = self.value
+        yield textual.containers.HorizontalGroup(
+            textual.widgets.Button("?", id="help"), textual.widgets.Static(self.info.name)
+        )
+        yield self.input_field
+        # To create extra spacing
+        yield textual.widgets.Static("")
+
+    @textual.on(textual.widgets.Button.Pressed, "#help")
+    def show_help(self):
+        self.app.push_screen(InfoScreen(self.info.description))
+
+
+class PluginSettingsTab(textual.widgets.Static):
+    __config_manager = surfactant.configmanager.ConfigManager()
+    # Core settings - I don't think there's a good way to automatically extract these
+    __setting = surfactant.utils.get_plugin_settings.PluginSetting
+    __core_settings = [
+        __setting(
+            "output_format",
+            "str",
+            "SBOM output format, see --list-output-formats for list of options",
+            "CyTRICS",
+        ),
+        __setting("recorded_institution", "str", "Name of user's institution.", "LLNL"),
```

Comment:
```suggestion
        __setting("recorded_institution", "str", "Name of user's institution.", ""),
```

Default recorded institution as empty string, since theoretically all external users will want something else.